Release Signer Alongside Node

BowTiedDevOps opened the pull request:
The aim of this PR is to create a signer release every time there is a core release happening, with the tag set as `<core_release>.0`.

Note: The [Actions PR](https://github.com/stacks-network/actions/pull/62) will have to be merged before this can be merged, as it references composite actions that are not existent in the current flow but in the one proposed by that PR.

**Example workflows (test failures are unaffected by these changes):**

- Node Release (built both `5.0.0.0.0` and `signer-5.0.0.0.0.0` releases): https://github.com/BowTiedDevOps/stacks-core/actions/runs/12953974660
- Standalone Signer Release (built `signer-5.0.0.0.0.1` release): https://github.com/BowTiedDevOps/stacks-core/actions/runs/12957741282
- `develop` branch CI run: https://github.com/BowTiedDevOps/stacks-core/actions/runs/12960606782

**Useful Links:**

- Releases Page: https://github.com/BowTiedDevOps/stacks-core/releases
- Stacks Core Docker Images: https://hub.docker.com/r/bowtieddevops/stacks-core/tags
- Stacks Signer Docker Images: https://hub.docker.com/r/bowtieddevops/stacks-signer/tags

wileyj commented:
Note that the required checks here will fail until the required actions PR is merged. 

@BowTiedDevOps let's also use this PR to fix the alpine images which are currently configured to use glibc (they should be using the musl binaries for the release) https://github.com/BowTiedDevOps/stacks-core/blob/develop/.github/actions/dockerfiles/Dockerfile.alpine-binary#L13-L16

CharlieC3 commented on .github/actions/dockerfiles/Dockerfile.alpine-binary:
```diff
@@ -10,10 +10,9 @@ ARG TARGETVARIANT
 ARG REPO
 
 RUN case ${TARGETPLATFORM} in \
-    linux/amd64/v2) BIN_ARCH=linux-glibc-x64-v2 ;; \
```
What's the context behind this platform being removed?

wileyj replied:
this change came from when i was confirming the images built in @BowTiedDevOps fork - what i noticed is that the alpine image is pulling in the glibc binaries (and has been for a few months - the alpine image either needs these changes, or needs to be removed since it's not being used). 

the `v2` variant is also not being built anymore (v3 is the default), so that match will never happen. 
it's possible we could build for that CPU in the future - but i would think that's unlikely since the arch is from 2008, and we haven't produced binaries for that arch in a long time. 


https://github.com/stacks-network/stacks-core/blob/develop/.github/workflows/create-source-binary.yml#L42

wileyj commented on .github/workflows/ci.yml:
```diff
@@ -92,7 +100,7 @@ jobs:
   ##  - it is not a release run
   docker-image:
     if: |
-      needs.check-release.outputs.is_release != 'true'
+      needs.check-release.outputs.is_signer_release != 'true'
```
i believe so - the problem we had previously when we tried to merge these changes is that it worked for a release build, but not for any other PR ci run. 

if `is_signer_release` is true, then `is_node_release` should also be true. if `is_signer_release` is false, we may assume that it's not a release workflow and we'll need to run the non-release jobs/steps. 

we may release the stacks-node + stacks-signer, or the stacks-signer, but not stacks-node by itself.

BowTiedDevOps replied:
Yes, that's correct - we have the possibility to run only signer-specific tests this way. I'll further add comments in the codebase to clarify this, but basically we have 3 cases:
- `is_signer_release` and `is_node_release` are both false - in this case, there's no release
- `is_signer_release` is true and `is_node_release` is false - this is a signer-only release, so we can run tests accordingly
- `is_signer_release` and `is_node_release` are both true - this is a node release (and implicitly a signer release as well, since we do both at the same time), and we can run node tests as well.

CharlieC3 replied:
Got it, thanks for confirming!

wileyj commented on .github/workflows/ci.yml:
```diff
@@ -155,9 +163,9 @@ jobs:
-  ##         - commit to either (development, next, master) branch
   stacks-core-build-tests:
     if: |
+      needs.check-release.outputs.is_signer_release == 'true' ||
```
agreed. @BowTiedDevOps  i think it's a good idea to add comments where this conditional is used so it's clear what triggers these jobs.

BowTiedDevOps replied:
Will do, thank you

BowTiedDevOps replied:
https://github.com/bowtieddevops/stacks-core/commit/3d676de19637ebc918575ac0c452f72a2dfe1ba4

CharlieC3 replied:
Much appreciated!

wileyj commented:
will double-check the binaries/images in the fork, otherwise it looks good to me (and should resolve the issue from the previous PR)

BowTiedDevOps commented:
I've updated the comments to further explain the outputs of the `check-release` job. I've also changed the conditionals to look like this:
```diff
+ needs.check-release.outputs.is_node_release == 'true' ||
  needs.check-release.outputs.is_signer_release == 'true'
```

Even though this is not necessary for the tests to run properly (i.e. `is_signer_release` will always be true if `is_node_release` is true), it adds more clarity to what the conditions do.

CharlieC3 reviewed (approved):
:shipit: LGTM! Thanks for the added commentary!

wileyj commented:
double checked the binaries/images after discovering we've had broken alpine docker images for a while. latest changes resolve problem

obycode reviewed (approved):
I think this looks good.

blockstack-devops commented:
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
